api/envoy/extensions/filters/http/tcp_post/v3/tcp_post.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.tcp_post.v3;
+
+import "envoy/config/route/v3/route_components.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.tcp_post.v3";
+option java_outer_classname = "TcpPostProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: TCP Post]
+// TCP Post :ref:`configuration overview <config_http_filters_tcp_post>`.
+// [#extension: envoy.filters.http.tcp_post]
+
+// Configuration for TcpPost filter.
+message TcpPost {
+  // Specifies a set of request headers to trigger converting POST requests
+  // to CONNECT and set ``protocol: bytestream`` header.
+  //
+  // This is mainly used by the upstream proxy to coordinate with the downstream
+  // proxy that uses HTTP POST for TCP proxying.
+  repeated config.route.v3.HeaderMatcher headers = 1;
```

Comment:
should we allow or reject changes to  : prefixed headers?
I'm inclined to say we should reject them, for consistency with other Envoy header modifiers unless you actually want to allow  host overrides, in which case we should explicitly comment and test.

Reply:
Can we try to intercept the generic matching support here? I think we have enough built that we should be ready to use it. @snowp should be your contact here and this will be a simple use case that we can make sure works end-to-end. Thank you!

Reply:
file removed.